projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php

```diff
@@ -142,6 +142,11 @@ function wpcom_add_my_account_item_to_profile_menu( $wp_admin_bar ) {
  * @return string Name of the admin bar class.
  */
 function wpcom_custom_wpcom_admin_bar_class( $wp_admin_bar_class ) {
+	// Honor the user's preference to hide the admin bar on the front end.
+	if ( ! is_admin() && get_user_option( 'show_admin_bar_front' ) !== 'true' ) {
```

Comment:
Hmm, this looks hacky. I would expect, if `get_user_option( 'show_admin_bar_front' )` is not `true` then the admin bar wouldn't show by internal WP mechanism. I will be surprised if devs need to explicitly add this check on `wp_admin_bar_class` filter 🤔 

Why is this broken only on Atomic but not Simple?

Reply:
For example, in Simple sites, we also add a filter for `wp_admin_bar_class` here: fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Snqzva%2Qone.cuc%3Se%3Q32q00946%23109-og, but the feature works on Simple Classic without having to add this check.

Reply:
In Core, this code seems to decide whether to show the `admin_bar` on front: https://github.com/WordPress/wordpress-develop/blob/5a30482419f1b0bcc713a7fdee3a14afd67a1bca/src/wp-includes/admin-bar.php#L1314-L1403. 
I suggest debugging which branch is executed to understand why the option is not being respected on Atomic: https://github.com/WordPress/wordpress-develop/blob/5a30482419f1b0bcc713a7fdee3a14afd67a1bca/src/wp-includes/admin-bar.php#L1362-L1380.

I also found `add_filter( 'show_admin_bar', '__return_true' )`: https://github.com/Automattic/jetpack/blob/ea4560860b3a7c8731c0c96e355231d329298fa8/projects/packages/masterbar/src/masterbar/class-masterbar.php#L185-L192, but (I think) it does not seem to be related to this issue, as it's not executed when `WPCOM_ADMIN_BAR_UNIFICATION` is true: https://github.com/Automattic/jetpack/blob/ea4560860b3a7c8731c0c96e355231d329298fa8/projects/packages/masterbar/src/class-main.php#L38

Reply:
It's weird. The setting works well on my atomic site without this change 🤔

Reply:
Really? It does not work on my Atomic Classic sites...

Reply:
https://github.com/user-attachments/assets/18f0f0ce-b99e-4653-947b-890ec09396b1

Reply:
> I also found add_filter( 'show_admin_bar', '__return_true' ): but (I think) it does not seem to be related to this issue

I saw that too. I tried removing it for good measure, but it doesn't seem to help. https://github.com/Automattic/jetpack/pull/38490/files

I think the value on Atomic is changing at this filter [here](https://github.com/WordPress/wordpress-develop/blob/5a30482419f1b0bcc713a7fdee3a14afd67a1bca/src/wp-includes/admin-bar.php#L1380). Otherwise, it's correct up to that point. However, I could not trace where or how that filter is being updated.

@arthur791004, this is a long shot, but what theme are you using on that site? Does this feature work on all of your Atomic sites?

Reply:
Turns out the Toolbar is forced to be visible while proxied by the debug-bar-loader plugin. So, we will opt to show a note to a12ns on the Profile page informing them of such. p1721787027964429/1721786928.757809-slack-C06DN6QQVAQ

Notice PR: https://github.com/Automattic/jetpack/pull/38519